### ISSUE TITLE
[Shaman] Maelstrom weapon wrong PPM value

### ISSUE THREAD

**Issue by Selot1234** [SunwellTracker/issues#2875]
Decription: the proc rate of shaman [buff](https://db.darkwizard.pl/?spell=51532) Maelstrom weapon (rank 5) is too low.

How it works: Current value is somewhere around 9 PPM. [link](http://www.worldoflogs.com/reports/epll49u2ree6vmn5/details/1/) to logs from Frosthold.
Total value of hits (with windfury, stormstrike, lava lash): 3117
Total gains of Maelstrom weapon buff: 1202. It gives 38.5%
Note [shaman](https://sunwell.pl/armory/Frosthold/Mazir) is using 2.60 speed weapons.

How it should work: 
Proc per hit with 2.60 speed weapon and 20 PPM mechanics should be ~ 87%
[formula](https://wowwiki.fandom.com/wiki/Procs_per_minute): PPH = Weapon speed * PPM / 60

Source (you should point out proofs of your report, please give us some source): 
https://db.darkwizard.pl/?spell=51532 -- 20 PPM
https://wowwiki.fandom.com/wiki/Maelstrom_Weapon -- approximately 25 PPM

**Comment by DollpartsFTP** [comment]
The proc rate should be 20% (1 proc per 5 attacks). This is why haste was primarily the best stat, as every 1% haste increased your proc rate by 1% effectively.

It's also why the Ulduar set bonus is possibly the greatest set bonus in WoW history, as it effectively doubles your chance to proc Maelstrom Weapon from 20% to 40% per swing.

**Comment by Selot1234** [comment]
@DollpartsFTP thanks for your insight.

Can you show your source of info, or refer to the proofs i have shown?

**Comment by DollpartsFTP** [comment]
Nearly impossible to find sources on things like this, that quite literally change every single expansion, and was changed 2-3 times in WOTLK alone, but keeps the same spell ID etc throughout. You can find some info on these boards, with "likes" by Blizzard employees at the time., but that's about as good as you're gonna get on this subject.

https://www.mmo-champion.com/threads/811901-does-maelstrom-weapon-have-an-internal-cooldown
https://forum.wow-freakz.com/FIXED-Shaman-Enhancement-Passive-Maelstrom-weapon-t348732.html
There are a few Blizzard forum posts, but can't make links for them. Ultimately either way this is impossible to prove.


The MMOC forum post being the only reliable source,
As a side note, dual-wielding does not change PPM.

For clarity, PPM functions as follows:
PPH is the chance for each hit (melee swing, melee ability, etc) to trigger a talent (in this case, MSW).
PPH = (WSpeed) / (60 / PPM)

Applied to MSW's 9 PPM for Enhancement Shamans, this amounts to:
PPH = 2.6 / (60 / 9) = 0.39
--> 39% chance for each hit to trigger MSW.


PPM itself is not modified by Haste, talents or anything else. This means Flurry does not hinder it.
Haste only makes you attack faster, which gives you more chances to proc.
-- 25 PPM would equal a 108% proc chance, which is obviously not possible, hence my comment about the Warmane forum people not knowing what they're talking about.

Your testing shows a total of 530 Swings and 40 LB casts. Assuming you only used them at 5 stacks (I know you did not), this makes it 200 MSW procs, or a 37.7% proc rate per attack.
Taking into account the 4 stacks you wasted, that is 204 procs, or a 38.4% proc rate.

Simcraft, with a profile modified to only auto-attack -- and use LB when it reaches 5 stacks of MSW -- for 6min on a Patchwerk fight, showed 388 Swings, 156 MSW procs, and 37 LB casts.
This amounts to 40.2% proc per attack.

The same profile, using the same parameters, and a more precise simulation (500k iterations), for 10min this time, showed 636 Swings, 257 MSW procs, and 49 LB casts.
This is a 40.4% MSW proc on each swing.

While Simcraft code seems to show 10 PPM (I checked), the proc rate obtained on simulations are not consistent with such a number.
9 PPM amounts to 39% per attack, 10 PPM would be 43.3%.

--> Based on said MMOC post, the proc rates you obtained are close to correct.
If you were to use abilities while testing, you would see a significant increase in procs.

TL;DR: If the information provided on the MMOC thread is correct (which is to be assumed, considering a Blizzard employee gave a positive comment on it), and MSW is indeed intended to be set at 9 PPM, your testing did not show a statistically significant difference in procs, and it seems MSW is working as -- or close to -- intended.

Also TL;DR - If it was 25PPM whilst duel weilding your proc % would be 109%., ridiculous, i think you'll agree.

**Comment by DollpartsFTP** [comment]
Important to note though, the above mathematics is only for melee swings.

Stormstrike, Windfury Weapon, Lava Lash (Added melee attacks, basically), all have the same rate of proccing Maelstrom as melee swings.

**Comment by Cwiercpasiec** [comment]
Here are some retail ICC logs:
http://www.worldoflogs.com/reports/b2k9jsgjwxc5urmd/log/?s=4091&e=4357
I checked Maelstrom procs a while ago, don't remember exact numbers, but they were more or less matching Sunwell's proc rate.

**Comment by Selot1234** [comment]
@Cwiercpasiec 
thx so much for these logs, they are rly valuable.

shammy on rotface back then had total 488 hits and 205 maelstrom weapon procs
this equals to 42% proc chance per hit and 10PPM assuming he was using 2.6 speed weapons.

The numbers are close to those on sunwell (38.5%, 9PPM).
